pkg/rbeconfigsgen/rbeconfigsgen.go

```diff
@@ -651,16 +651,18 @@ func processTempDir(o *Options) error {
 // genConfigBuild generates the contents of a BUILD file with a toolchain target pointing to the
 // C++ toolchain related rules generated by Bazel and a default platforms target.
 func genConfigBuild(o *Options) (generatedFile, error) {
-	if o.PlatformParams.CppToolchainTarget != "" {
-		return generatedFile{}, fmt.Errorf("<internal error> C++ toolchain target was already set")
-	}
 	// Populate the C++ toolchain target if C++ config generation is enabled.
 	if o.GenCPPConfigs {
-		o.PlatformParams.CppToolchainTarget = "//cc:cc-compiler-k8"
+		if o.CPPToolchainTarget != "" {
+			o.PlatformParams.CppToolchainTarget = fmt.Sprintf("//cc:%s", o.CPPToolchainTarget)
+		}
 		if o.OutputConfigPath != "" {
-			o.PlatformParams.CppToolchainTarget = fmt.Sprintf("//%s/cc:cc-compiler-k8", path.Clean(o.OutputConfigPath))
+			// The Toolchain target should always start with `//`.
+			o.PlatformParams.CppToolchainTarget = strings.TrimPrefix(o.PlatformParams.CppToolchainTarget, "/")
+			o.PlatformParams.CppToolchainTarget = fmt.Sprintf("//%s%s", path.Clean(o.OutputConfigPath), o.PlatformParams.CppToolchainTarget)
 		}
 	} else {
+		o.PlatformParams.CppToolchainTarget = ""
```

Comment:
I think the cpp toolchain target path construction logic has become complicated enough that we should refactor it into its own function and add unit tests. Specifically I'm interested in making sure the target path is constructed correctly for various values of `o.OutputConfigPath`

Reply:
Done.